packages/core/database/lib/query/query-builder.js

```diff
@@ -302,9 +302,15 @@ const createQueryBuilder = (uid, db) => {
         }
         case 'count': {
           const dbColumnName =
-            state.count === '*' ? '*' : this.aliasColumn(helpers.toColumnName(meta, state.count));
-
-          qb.count({ count: dbColumnName });
+            state.count === '*'
+              ? this.aliasColumn(helpers.toColumnName(meta, 'id'))
+              : this.aliasColumn(helpers.toColumnName(meta, state.count));
+
+          if (this.shouldUseDistinct(2)) {
```

Comment:
why do you need to have at least 2 joins?

Reply:
The counting issue is really only present when we have multiple deep filters, and so considering as you said before that count distinct has a big impact on performance I wanted to make sure that we only use it when necessary essentially

Reply:
as soon as you make a join you might have duplicated rows 🤔 deep filtering can imply only one join and still create problems so I would keep the old distinct here :)

Reply:
ok sounds good @alexandrebodin